src/backend/distributed/transaction/backend_data.c

```diff
@@ -634,6 +636,8 @@ UnSetDistributedTransactionId(void)
 		MyBackendData->transactionId.timestamp = 0;
 
 		SpinLockRelease(&MyBackendData->mutex);
+
+		UnmarkCitusInitiatedBackend();
```

Comment:
Can't we do that while the spinlock is already held? 

I think we can introduce a function called `UnmarkCitusInitiatedBackendNoLock()` which doesn't acquire the lock and relies on the called holding the locks.

We could even not have the function and mark the field itself directly.